lib/Dialect/Arith/Conversions/ArithToCGGI/ArithToCGGI.cpp

```diff
@@ -51,15 +51,53 @@ static bool allowedRemainArith(Operation *op) {
             }
             return false;
           })
+      .Case<mlir::arith::SubIOp, mlir::arith::AddIOp, mlir::arith::MulIOp>(
+          [](auto op) {
+            // This lambda will be called for any of the matched operation types
+            if (auto lhsDefOp = op.getOperand(0).getDefiningOp()) {
+              auto lshAllowed = allowedRemainArith(lhsDefOp);
+              if (auto rhsDefOp = op.getOperand(1).getDefiningOp()) {
+                auto rhsAllowed = allowedRemainArith(rhsDefOp);
+                return lshAllowed && rhsAllowed;
+              }
+            }
+            return false;
+          })
       .Default([](Operation *) {
         // Default case for operations that don't match any of the types
         return false;
       });
 }
 
 static bool hasLWEAnnotation(Operation *op) {
-  return static_cast<bool>(
-      op->getAttrOfType<mlir::StringAttr>("lwe_annotation"));
+  auto check =
+      static_cast<bool>(op->getAttrOfType<mlir::StringAttr>("lwe_annotation"));
```

Comment:
I think this returns nullptr if it's not found, so you could just store it as a StringAttr and then compare it (implicitly) to nullptr. Same as below